### PR TITLE
[Button] Remove connectedDisclosure

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/10126

### WHAT is this pull request doing?

Removes all code related to connectedDisclosure
Updates the split example to use Popover, ActionList, and ButtonGroup

### CODE REVIEW

**Review by sam-b-rose — COMMENTED** [review]
Code removal looks great! I think this may just need to be rebased after [this Button change](https://github.com/Shopify/polaris/pull/10164).

File: polaris-react/src/components/Button/Button.scss [hunk 1]
```diff
@@ -197,7 +175,7 @@
 }
 
 .primary,
-.destructive,
+.critical,
```
Comment (sam-b-rose) [inline polaris-react/src/components/Button/Button.scss:178]: Thanks for catching and updating this! 👍

**Review by sam-b-rose — APPROVED** [review]
Nice Button clean up 🧽 ✨

**Review by aveline — APPROVED** [review]
Code looks good, there's a slight change in the Chromatic diff though.

**Comment by kyledurand** [comment]
> Code looks good, there's a slight change in the Chromatic diff though.

That's intentional, the size of segmented buttons was bigger than it should have been because of the extra styles. These will now match with other inline button sizes

**Pull request merged**